### PR TITLE
Cplat cloudservice2

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description

adding two new parameter sets for New-AzCloudService, which will create a Cloud Service resource using information passed in -configurationFile and -definitionFile 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by grizzlytheodore** [comment]
I dont see changelog in CloudService folder.  Where do it put that?

**Comment by grizzlytheodore** [comment]
I'm not sure why Analyze tests are failing with errors from all the md files: 
Online version in the header of the file is incorrect. The corresponding URL should be the following: 
                https://docs.microsoft.com/powershell/module/az.cloudservice validation/get-azcloudservice

validation is a function I added in custom/New-AzCloudService.ps1 to validate all the configuration.

**Comment by BethanyZhou** [comment]
> I dont see changelog in CloudService folder. Where do i put that?

Please see https://github.com/Azure/azure-powershell/blob/main/src/CloudService/Changelog.md

File: src/CloudService/custom/readme.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ These provide functionality to our HTTP pipeline and other useful features. In s
 ### Attributes
 For processing the cmdlets, we've created some additional attributes:
 - `Microsoft.Azure.PowerShell.Cmdlets.CloudService.DescriptionAttribute`
-  - Used in C# cmdlets to provide a high-level description of the cmdlet. This is propegated to reference documentation via [help comments](https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_comment_based_help) in the exported scripts.
+  - Used in C# cmdlets to provide a high-level description of the cmdlet. This is propegated to reference documentation via [help comments](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comment_based_help) in the exported scripts.
```
Comment (BethanyZhou) [inline src/CloudService/custom/readme.md:35]: Please revert the change for this file

File: src/CloudService/custom/New-AzCloudService.ps1 [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+.Synopsis
+Create a CloudService Resource
+.Description
+Create a CloudService Resource 
+.Link
+https://docs.microsoft.com/powershell/module/az.cloudservice/new-azcloudservice
+#>
+
+function New-AzCloudService {
+    [OutputType('Microsoft.Azure.PowerShell.Cmdlets.CloudService.Models.Api20210301.ICloudService')]
```
Comment (BethanyZhou) [inline src/CloudService/custom/New-AzCloudService.ps1:26]: Hey, I think there is a better way to implement New-AzCloudService rather than handcraft whole cmdlet.
- First, change the branch to aa62339078c075d81aa1019c62450805f750e2d7
image
image
- Re-generate a cmdlet then you can find a raw generated  New-AzCloudService
- Adapt generated cmdlet by customization.

Comment (grizzlytheodore) [reply src/CloudService/custom/New-AzCloudService.ps1:26]: so in this PR I am adding two new parameter sets to New-AzCloudService. The default parameter set is generated from Swagger, but the two new parameter sets have a lot more functionality than what we could generate from swagger. it takes in cloudService configurations from CsCfg, CsDef, and CsPkg files and validates that all the configuration is sound. It also creates other resources if they're not already created, which before it required them to have those resources created. 

pretty much, we only added here to customization what could not be generated from swagger.

**Comment by grizzlytheodore** [comment]
> > I dont see changelog in CloudService folder. Where do i put that?
> 
> Please see https://github.com/Azure/azure-powershell/blob/main/src/CloudService/Changelog.md

I see that that's in the main branch, but I don't see changelog files in any of the PRs in the 'generation' branch for RPs that are generated

**Comment by BethanyZhou** [comment]
> > > I dont see changelog in CloudService folder. Where do i put that?
> > 
> > 
> > Please see https://github.com/Azure/azure-powershell/blob/main/src/CloudService/Changelog.md
> 
> I see that that's in the main branch, but I don't see changelog files in any of the PRs in the 'generation' branch for RPs that are generated

Yes, we will fill change log when we sync generation to main branch. We will take care of that job Could you comment the change log msg here and then we will add them as change log.

@wyunchi-ms , could you help to review this PR?

**Review by wyunchi-ms — APPROVED** [review]
